Backend/src/config/passport-setup.ts

```diff
@@ -0,0 +1,61 @@
+import passport from 'passport';
+import { Strategy as GoogleStrategy } from 'passport-google-oauth20';
+import { type users } from '@prisma/client'; // Make sure to import the User type from Prisma
+import keys from './keys';
+import UserService from '@/modules/users/users.service';
+
+passport.serializeUser((user: users, done) => {
+  done(null, user.id);
+});
+
+passport.deserializeUser(async (id: string, done) => {
+  try {
+    const userService = new UserService();
+    const user = await userService.findUserById(id);
+
+    if (user) {
+      done(null, user);
+    } else {
+      done(new Error('User not found'), null);
+    }

```

Comment:
I think if user doesn't exist, findUserById will throw an error (not found error). so instead you can try/catch the error and call `done`. 

Please experiment it using non-exist user id.